src/Plugin.php

```diff
@@ -58,12 +59,41 @@ public function activate(Composer $composer, IOInterface $io)
      */
     public static function getSubscribedEvents()
     {
+        if (version_compare(PluginInterface::PLUGIN_API_VERSION, '2.0', 'lt')) {
+            // Events for Composer 1.
+            return [
+                PackageEvents::PRE_PACKAGE_INSTALL => 'prePackageInstall',
+                PackageEvents::PRE_PACKAGE_UPDATE => 'prePackageUpdate',
+            ];
+        }
+        // Events for Composer 2.
         return [
-            PackageEvents::PRE_PACKAGE_INSTALL => 'prePackageInstall',
-            PackageEvents::PRE_PACKAGE_UPDATE => 'prePackageUpdate',
+            PluginEvents::PRE_POOL_CREATE => 'prePoolCreate',
```

Comment:
See https://github.com/composer/composer/blob/master/UPGRADE-2.0.md#detailed-differences-in-event-flow-during-dependency-resolution-composer-updates-and-installs that describes event flow in Composer 2.